contracts/neutron-interchain-txs/src/contract.rs

```diff
@@ -0,0 +1,472 @@
+        interchain_account_id,
+        connection_id,
+    };
+
+    let res: QueryInterchainAccountAddressResponse = deps.querier.query(&query.into())?;
+    Ok(to_binary(&res)?)
+}
+
+pub fn query_interchain_address_contract(
+    deps: Deps<InterchainQueries>,
+    env: Env,
+    interchain_account_id: String,
+) -> ContractResult<Binary> {
+    let key =
+        "icacontroller-".to_string() + env.contract.address.as_str() + "." + &interchain_account_id;
+    let address = INTERCHAIN_ACCOUNTS.load(deps.storage, key)?;
+
+    Ok(to_binary(&address)?)
+}
+
+pub fn get_next_id(store: &mut dyn Storage) -> StdResult<u64> {
+    REPLY_QUEUE_ID
+        .keys(store, None, None, cosmwasm_std::Order::Descending)
+        .next()
+        .unwrap_or(Ok(IBC_SUDO_ID_RANGE_START))
+        .map(|id| id + 1)
+}
+
+pub fn save_reply_payload(store: &mut dyn Storage, payload: SudoPayload) -> StdResult<u64> {
+    let id = get_next_id(store)?;
+    REPLY_QUEUE_ID.save(store, id, &to_vec(&payload)?)?;
+    Ok(id)
+}
+
+pub fn read_reply_payload(store: &mut dyn Storage, id: u64) -> StdResult<SudoPayload> {
+    let data = REPLY_QUEUE_ID.load(store, id)?;
+    from_binary(&Binary(data))
+}
+
+pub fn read_sudo_payload(store: &mut dyn Storage, seq_id: u64) -> StdResult<SudoPayload> {
+    let data = SUDO_PAYLOAD.load(store, seq_id)?;
+    from_binary(&Binary(data))
+}
+
+fn msg_with_sudo_callback<C: Into<CosmosMsg<T>>, T>(
+    deps: DepsMut,
+    msg: C,
+    payload: SudoPayload,
+) -> StdResult<SubMsg<T>> {
+    let id = save_reply_payload(deps.storage, payload)?;
+    Ok(SubMsg::reply_on_success(msg, id))
+}
+
+fn execute_register_ica(
+    deps: DepsMut,
+    env: Env,
+    connection_id: String,
+    interchain_account_id: String,
+) -> StdResult<Response<NeutronMsg>> {
+    let register =
+        NeutronMsg::register_interchain_account(connection_id, interchain_account_id.clone());
+    let key =
+        "icacontroller-".to_string() + env.contract.address.as_str() + "." + &interchain_account_id;
+    INTERCHAIN_ACCOUNTS.save(deps.storage, key, &None)?;
+    Ok(Response::new().add_message(register))
+}
+
+fn execute_delegate(
+    mut deps: DepsMut,
+    env: Env,
+    interchain_account_id: String,
+    validator: String,
+    amount: u128,
+) -> StdResult<Response<NeutronMsg>> {
+    let key =
+        "icacontroller-".to_string() + env.contract.address.as_str() + "." + &interchain_account_id;
+
+    let (delegator, connection_id) = INTERCHAIN_ACCOUNTS
+        .load(deps.storage, key)?
+        .ok_or_else(|| StdError::generic_err("Interchain account is not created yet"))?;
+
+    let delegate_msg = MsgDelegate {
+        delegator_address: delegator,
+        validator_address: validator,
+        amount: Some(Coin {
+            denom: "stake".to_string(),
+            amount: amount.to_string(),
+        }),
+    };
+    let mut buf = Vec::new();
+    buf.reserve(delegate_msg.encoded_len());
+
+    if let Err(e) = delegate_msg.encode(&mut buf) {
```

Comment:
i have this weird thing in my IDE here:

<img width="719" alt="image" src="https://user-images.githubusercontent.com/6124982/184311646-9500a838-3917-4f2a-8196-aacaaa92dcef.png">

Reply:
I have the same thing but seems like it's some IDE bug. Everything is fine during compilation

Reply:
#1 Rustaceans' rule: never trust the IDE

Reply:
just use vim lmao :sunglasses: